apps/condo/domains/miniapp/schema/B2BAppAccessRightSet.test.js

```diff
@@ -0,0 +1,650 @@
+            })
+
+            test('Can not hard-delete', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(admin, app)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await B2BAppAccessRightSet.delete(notServiceUser, createdAccessRightSet.id)
+                })
+            })
+        })
+
+        describe('Anonymous', () => {
+            test('Can not create', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await createTestB2BAppAccessRightSet(anonymous, app)
+                })
+            })
+
+            test('Can not read', async () => {
+                await expectToThrowAuthenticationErrorToObjects(async () => {
+                    await B2BAppAccessRightSet.getAll(anonymous, {
+                        id: accessRightSet.id,
+                    })
+                })
+            })
+
+            test('Can not update', async () => {
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestB2BAppAccessRightSet(anonymous, accessRightSet.id)
+                })
+            })
+
+            test('Can not soft-delete', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(admin, app)
+
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await B2BAppAccessRightSet.softDelete(anonymous, createdAccessRightSet.id)
+                })
+            })
+
+            test('Can not hard-delete', async () => {
+                const [app] = await createTestB2BApp(admin)
+                const [createdAccessRightSet] = await createTestB2BAppAccessRightSet(admin, app)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await B2BAppAccessRightSet.delete(anonymous, createdAccessRightSet.id)
+                })
+            })
+        })
+    })
+
+    describe('Constraints', () => {
+        test('Cannot be created 2 links for a app', async () => {
+            const [app] = await createTestB2BApp(admin)
+            await createTestB2BAppAccessRightSet(admin, app)
+
+            await expectToThrowUniqueConstraintViolationError(async () => {
+                await createTestB2BAppAccessRightSet(admin, app)
+            }, 'b2b_app_access_right_set_unique_app')
+        })
+    })
+})
+
+describe('B2BApp permissions for service user', () => {
+    let admin
+    let user
+    let serviceUserClient
+    let serviceUser
+
+    beforeEach(async () => {
+        admin = await makeLoggedInAdminClient()
+        user = await makeClientWithNewRegisteredAndLoggedInUser()
+        const [newServiceUser] = await registerNewServiceUserByTestClient(admin)
+        serviceUser = newServiceUser
```

Comment:
How about to assign user to `.user` prop to be more consistent with all project tests?